frontend/components/MoviePopup/MoviePopup.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import Image from 'next/image';
+import cn from 'classnames';
+import ReactPlayer from 'react-player';
+import { useAppSelector, useAppDispatch } from 'store/hooks';
+import { actions as moviePreviewActions } from 'features/moviePreview';
+import { PayloadAction } from '@reduxjs/toolkit';
+import { Element, ElementType, Genre, ProductionCountry } from 'types/MovieAPI';
+import { movieFetcher } from 'api/api';
+import { Error } from 'components/Error';
+import { AiOutlinePlus } from 'react-icons/ai';
+import { HiOutlineX } from 'react-icons/hi';
+import { BiVolumeFull, BiVolumeMute } from 'react-icons/bi';
+import { GoThumbsup } from 'react-icons/go';
+import playIcon from 'images/play.svg';
+import styles from 'components/MoviePopup/MoviePopup.module.scss';
+
+const MoviePopup: React.FC = (): ReactElement => {
+  const [trailer, setTrailer] = useState<string>('');
+  const [genres, setGenres] = useState<Genre[]>([]);
+  const [isVolumeMuted, setIsMuted] = useState<boolean>(false);
+  const [productionCountries, setProductionCountries] = useState<ProductionCountry[]>([]);
+
+  const dispatch = useAppDispatch();
+  const isMoviePopupVisible = useAppSelector((state) => state.moviePreview.isMoviePopupVisible);
+  const previewMovie = useAppSelector((state) => state.moviePreview.movieForPreview);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>): void => {
+    if (event.key === 'Escape') {
+      dispatch(moviePreviewActions.closeMoviePopup());
+    }
+  }
+
+  useEffect(() => {
+    if (!previewMovie) return;
+
+    async function fetchMovie(): Promise<void> {
+      const { data } = await movieFetcher.get(
+        `/${
```

Comment:
why you did not make this request in getServerSideProps?

Reply:
that's because i need the data about the chosen movie from the store and react hooks cannot be used outside of react function components or custom react hooks